src/components/DateRangeSelector.tsx

```diff
@@ -0,0 +1,136 @@
+  const today = new Date();
+  const [startDate, setStartDate] = useState<Date>(dateFrom ? dateFrom : minimumDate);
+  const [endDate, setEndDate] = useState<Date>(dateTo ? dateTo : today);
+  const exploreUrl = useExploreUrl();
+
+  useEffect(() => {
+    const { dateFrom, dateTo } = dateRangeToDates(dateRange);
+    const startDate = dateFrom ? dateFrom : minimumDate;
+    const endDate = dateTo ? dateTo : today;
+    setStartDate(startDate);
+    setEndDate(endDate);
+
+    return () => {};
+    // eslint-disable-next-line
+  }, [dateRange]);
+
+  const handleStartDateChange = (date: Date) => {
+    const formattedDateRange = formatDateRangeForUrl(date, dateTo);
+    if (formattedDateRange) exploreUrl?.setDateRange(formattedDateRange);
+    setStartDate(date);
+  };
+
+  const handleEndDateChange = (date: Date) => {
+    const formattedDateRange = formatDateRangeForUrl(dateFrom, date);
+    if (formattedDateRange) exploreUrl?.setDateRange(formattedDateRange);
+    setEndDate(date);
+  };
+
+  const formatDateRangeForUrl = (dateFrom?: Date, dateTo?: Date): DateRange | undefined => {
+    // e.g:  from=2021-06-22&to=2021-09-22
+    const from = dateFrom ? dateFrom : minimumDate;
+    const to = dateTo ? dateTo : today;
+    return `from=${dayjs(from).format('YYYY-MM-DD')}&to=${dayjs(to).format('YYYY-MM-DD')}` as DateRange;
+  };
+
+  const handleStartDateRaw = (event: React.FocusEvent<HTMLInputElement>) => {
+    if (event && event.target && event.target.value && dateStringRegex.test(event.target.value)) {
+      handleStartDateChange(new Date(event.target.value));
+    }
+  };
+
+  const handleEndDateRaw = (event: React.FocusEvent<HTMLInputElement>) => {
+    if (event && event.target && event.target.value && dateStringRegex.test(event.target.value)) {
+      handleEndDateChange(new Date(event.target.value));
+    }
+  };
+
+  const handleStartDateMonthChange = (date: Date) => {
+    setStartDate(date);
+  };
+
+  const handleEndDateMonthChange = (date: Date) => {
+    setEndDate(date);
+  };
+
+  const handleStartDateSelect = (date: Date, event: React.SyntheticEvent<any> | undefined) => {
+    if (event) {
+      handleStartDateChange(date);
+    }
+  };
+
+  const handleEndDateSelect = (date: Date, event: React.SyntheticEvent<any> | undefined) => {
+    if (event) {
+      handleEndDateChange(date);
+    }
+  };
+
+  return (
+    <>
+      <div className='flex flex-wrap space-x-1.5'>
+        <div className='flex flex-row items-end inline-block align-middle'>
+          <HeaderDateRangeSelect exploreUrl={exploreUrl} />
+        </div>
+        <div className='flex flex-nowrap space-x-1.5'>
+          <div className='flex flex-row items-end inline-block align-middle'>
+            <ReactDatePicker
+              enableTabLoop={false}
+              disabledKeyboardNavigation={true}
+              preventOpenOnFocus={true}
+              className='border rounded py-1.5 px-3 w-32 focus:outline-none focus:ring focus:border-blue-200'
```

Comment:
@TKGZ This is the input style, maybe you can play around with it to see how the input won't get cutoff on your screen?

Reply:
You should just get rid of all right padding.

The problem seems to be: the width is fixed to xREM, when it should be adjustd to length of text.

Plus: you have right padding which in case of too narrow box will cut off text from the right.

So set right padding to 0. And then possibly make the box width better adjusted. Min x but possibly longer if necessary.

Reply:
But how come it appears normal on my screen than yours? I am just puzzled but this difference....